src/Microsoft.Diagnostics.Monitoring.EventPipe/Logs/TraceEventExtensions.cs

```diff
@@ -26,21 +26,22 @@ public static void GetLogMessageJsonEventData(
             string argsJson = (string)traceEvent.PayloadValue(6);
             string formattedMessage = (string)traceEvent.PayloadValue(7);
 
-            string? activityTraceId;
-            string? activitySpanId;
-            string? activityTraceFlags;
-            if (traceEvent.Version >= 2)
+            string? activityTraceId = null;
+            string? activitySpanId = null;
+            string? activityTraceFlags = null;
+
+            int idxTraceId = traceEvent.PayloadIndex("ActivityTraceId");
+            int idxSpanId = traceEvent.PayloadIndex("ActivitySpanId");
+            if (idxTraceId >= 0 && idxSpanId >= 0)
             {
-                // Note: Trace correlation fields added in .NET 9
-                activityTraceId = (string)traceEvent.PayloadValue(8);
-                activitySpanId = (string)traceEvent.PayloadValue(9);
-                activityTraceFlags = (string)traceEvent.PayloadValue(10);
+                activityTraceId = (string)traceEvent.PayloadValue(idxTraceId);
+                activitySpanId = (string)traceEvent.PayloadValue(idxSpanId);
             }
-            else
+
+            int idxFlags = traceEvent.PayloadIndex("ActivityTraceFlags");
+            if (idxFlags >= 0)
```

Comment:
This is fine but should move this inside the `if (idxTraceId >= 0 && idxSpanId >= 0)` block? I am seeing the code producing these values either produce all or not.